### PR TITLE
LPS-188239 Update Object Relationships while updates Object Definition

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://liferay.atlassian.net/browse/LPS-188239

### CODE REVIEW

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java [hunk 1]
```diff
@@ -675,6 +671,32 @@ private void _addObjectDefinitionResources(
...
+				objectRelationships, ObjectRelationship::getName);
+
+			List<String> oldObjectRelationshipsNames = transform(
+				_objectRelationshipLocalService.getObjectRelationships(
+					objectDefinitionId),
+				com.liferay.object.model.ObjectRelationship::getName);
+
+			Set<String> objectRelationshipsToBeDeleted =
+				SetUtil.asymmetricDifference(
+					oldObjectRelationshipsNames, objectRelationshipsNames);
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java:684]: @mateussgn rename to `objectRelationshipNamesToBeDeleted`

can we do this?

```java
Set<String> objectRelationshipNamesToBeDeleted =
		SetUtil.asymmetricDifference(
			transform(
				_objectRelationshipLocalService.getObjectRelationships(
					objectDefinitionId),
				com.liferay.object.model.ObjectRelationship::getName), 
			transform(objectRelationships, ObjectRelationship::getName));
```

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java [hunk 2]
```diff
@@ -675,6 +671,32 @@ private void _addObjectDefinitionResources(
...
+
+			for (String objectRelationshipToBeDeleted :
+					objectRelationshipsToBeDeleted) {
+
+				com.liferay.object.model.ObjectRelationship
+					oldObjectRelationship =
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java:690]: @mateussgn can we rename to objectRelationship here? 

we just need to define a preffix in a variable when we have conflicts, in this case I believe we don't have conflicts.

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java [hunk 3]
```diff
@@ -684,13 +706,29 @@ private void _addObjectDefinitionResources(
 				).build();
 
 			for (ObjectRelationship objectRelationship : objectRelationships) {
+				String objectRelationshipName = objectRelationship.getName();
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java:709]: @mateussgn we don't need this variable, we can just keep calling the method `objectRelationship.getName();`

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java [hunk 4]
```diff
@@ -684,13 +706,29 @@ private void _addObjectDefinitionResources(
...
+
+				com.liferay.object.model.ObjectRelationship
+					oldObjectRelationship =
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java:712]: @mateussgn rename to originalObjectRelationship to keep the modelListener pattern, [see](https://github.com/liferay/liferay-portal/blob/6ce0b9eb049e99a50285a324a3bd4afd76d340ab/modules/apps/object/object-service/src/main/java/com/liferay/object/internal/model/listener/ObjectRelationshipModelListener.java#L48)

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:object

**Review by MarinhoFeliphe — APPROVED** [review]

**Comment by MarinhoFeliphe** [comment]
LGTM!! :tada:

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:forward

**Comment by mateussgn** [comment]
ci:forward:force

**Pull request closed without merging**